frontend/src/app/modules/projects/projects-page/projects-page.component.ts

```diff
@@ -0,0 +1,54 @@
+import { Component, OnInit } from '@angular/core';
+import { MatDialog } from '@angular/material/dialog';
+import { NotificationService } from '@core/services/notification.service';
+import { ProjectService } from '@core/services/project.service';
+import { CreateProjectModalComponent } from '@modules/projects/create-project-modal/create-project-modal.component';
+import { Subject, takeUntil } from 'rxjs';
+
+import { ProjectDto } from '../../../models/projects/project-dto';
+
+@Component({
+    selector: 'app-projects-page',
+    templateUrl: './projects-page.component.html',
+    styleUrls: ['./projects-page.component.sass'],
+})
+export class ProjectsPageComponent implements OnInit {
+    projects: ProjectDto[] = [];
+
+    private unsubscribe$ = new Subject<void>();
+
+    constructor(
+        public dialog: MatDialog,
+        private projectService: ProjectService,
+        private notificationService: NotificationService,
+        // eslint-disable-next-line no-empty-function
+    ) {}
+
+    ngOnInit(): void {
+        this.loadProjects();
+    }
+
+    loadProjects(): void {
+        this.projectService.getAllProjects().pipe(
+            takeUntil(this.unsubscribe$),
+        ).subscribe(
+            (projects: ProjectDto[]) => {
+                this.projects = projects;
+            },
+            () => {
+                this.notificationService.error('Failed to load projects');
+            },
+        );
+    }
+
+    openCreateModal(): void {
+        const dialogRef = this.dialog.open(CreateProjectModalComponent, {
+            width: '35%',
+            height: '45%',
```

Comment:
Width in % can be really bad idea if you open on big or really small screen
Partially it's true for height